### ISSUE TITLE
Detekcija i određivanje rase pasa na slikama

### ISSUE THREAD

**Issue by rohirim91** [ftn-ai-lab/sc-2020-siit#7]
**Tim**
Dragana Grbić, sw22/2017
Petar Nikolić, sw31/2017

**Asistenti**
Dragan Vidaković,
Milica Škipina

**Repo**
[detekcija-klasifikacija-pasa](https://github.com/rohirim91/soft-kompjuting---detekcija-i-klasifikacija-pasa)

**Definicija problema**
Cilj projekta je detekcija pasa na slikama i određivanje njihove rase. Ulaz u sistem predstavlja slika na kojoj se nalaze jedan ili više pasa. Izlaz iz sistema je sika na kojoj su psi uokvireni okvirom sa labelom koja predstavlja rasu psa.

**Motivacija zašto vredi rešavati dati problem**
Ovakav sistem može biti iskorišćen kao proof of concept za veći sistem koji bi obavljao određivanje rase pasa i mačaka i vrste papagaja i drugih kućnih ljubimaca. Takođe, mogao bi se, u unapređenom obliku, koristiti i prilikom lova ili safarija, zatim za potrebe snimanja dokumentarnih filmova kada je potrebno sa određene distance pronaći životinju koja je snimateljima od interesa.

**Skup podataka**
Na [linku ](http://vision.stanford.edu/aditya86/ImageNetDogs/) se nalazi skup podataka koji bismo koristili. Skup sadrži slike 120 rasa pasa sa oko 150 slika po rasi.
S obzirom na skroman broj slika, koristili bismo augmentaciju podataka. Sa slika bi primenom Selective Search i Intersect Over Union našli isečke slike koji se u velikoj meri poklapaju sa regionima slike na kojima se nalaze psi (sadržani u anotacijama slika).
Od tih isečaka bismo transformacijama slika (pomeraj, rotacija, flipovanje...) dobili još slika.
Takođe, oni isečci koji nemaju dovoljno veliko preklapanje sa reigonima slike na kojima su psi bi takođe bili ubačeni u skup slika.
Uz to, bilo bi ubačeno i par stotina slika ljudskih lica kako bismo model učinili što otpornijim na false pozitive.
Takođe, imajući u vidu skroman hardver koji imamo na raspolaganju, te nemogućnost treniranja veoma dubokih mreža, kao i velike sličnosti medju mnogim rasama (haski, malamut itd), za potrebe ovog projekta bi smo prepoznavali pse 5 rasa koje bismo odabrali tako da imaju međusobno dovoljno velike razlike u izgledu.

**Metodologija**
Usled činjenice da se na jednoj slici može naći više pasa, kao i to da bi na slikama sam pas mogao zauzimati samo mali deo slike, bilo bi potrebno koristiti neku region proposal tehniku. Mi bismo koristili Selective Search, ali bismo isprobali i neku složeniju implementaciju Sliding Window tehnike.
Za samu detekciju i određivanje rase bismo koristili custom-made konvolucionu neuronsku mrežu, ali bismo istražili da li postoje neki već istrenirani modeli koji bi barem mogli prepoznati pse na slici, dok bi odredjivanje rase radili nekim drugim pretreniranim modelom ili našim modelom.
Radi otklanjanja viška detekcija (jedan pas detektovan više puta), koristili bismo Non Maxima Suppression.
Radi bržeg treniranja mreža koristili bismo tensorflow podršku za korišćenje grafičke kartice.

**Evaluacija**
Goreopisani skup podataka bismo razdvojili na trening skup i validacioni skup u odosu 90:10. Testni skup bi se sastojao od po desetak slika svake rase koje sadrže više pasa na slici, a koje bi bile izdvojene iz polaznog skupa slika pre njegove augmentacije. Nakon treniranja modela, na test skupu možemo možemo odrediti efikasnost modela. To bismo radili računanjem procenta detektovanih pasa na slikama, procentom tačno određenih rasa i brojem false pozitiva. Prilikom treniranja mreže, od instersa bi nam bili validation accuracy i validation loss.

**Comment by vdragan1993** [comment]
Tema odobrena.
Ostavite link ka GitHub repozitorijumu projekta.
Srećan rad.
